### PR TITLE
[Modules] Updated EventGrid Topics to new dependencies approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated EventGrid Topics to new dependencies approach

| Pipeline |
| - |
| [EventGrid: Topics](https://github.com/Azure/ResourceModules/actions/workflows/ms.eventgrid.topics.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: utilities/tools/REST2CARML/Set-ModuleFolderData.ps1 [hunk 1]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (eriqua) [inline utilities/tools/REST2CARML/Set-ModuleFolderData.ps1:1]: 👻

Comment (MrMCake) [reply utilities/tools/REST2CARML/Set-ModuleFolderData.ps1]: removed

File: modules/Microsoft.EventGrid/topics/.test/pe/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,56 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/pe/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.EventGrid/topics/.test/pe/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.eventgrid.topics-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/pe/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.EventGrid/topics/.test/pe/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/pe/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.EventGrid/topics/.test/min/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,37 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.EventGrid/topics/.test/min/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.eventgrid.topics-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/min/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.EventGrid/topics/.test/min/deploy.test.bicep [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/min/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep [hunk 8]
```diff
@@ -0,0 +1,91 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.eventgrid.topics-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep (hunk 8 above)
Comment (eriqua) [inline modules/Microsoft.EventGrid/topics/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**